### ISSUE TITLE
Replace the `requests` dependency with `scrapy.mail`

### ISSUE THREAD

**Issue by CarterPape** [CarterPape/NewsBot#5]
There is an opportunity to replace the sole dependency on `requests` in [`item_emailer.py`](https://github.com/CarterPape/NewsBot/blob/master/src/newsbot/item_pipelines/item_emailer.py) with [`scrapy.mail`](https://docs.scrapy.org/en/latest/topics/email.html).

As documented in [an issue](https://github.com/scrapy/scrapy/issues/3478#issuecomment-731716070) on the Scrapy project, my previous foibles with `scrapy.mail` were the result of failing to see that `scrapy.mail.MailSender().send` returns a `twisted.internet.deferred.Deferred` (because this behavior is not documented).

An item pipeline can return a `Deferred`, though I don't see documented what happens when this is the case. Does the `Deferred` get immediately passed to the next item pipeline? Is the next pipeline called only after the `Deferred` finishes? Either way, how is the `item` passed along to the next pipeline?

Once I test this and find out, I could add both of these items to the Scrapy documentation.

**Comment by CarterPape** [comment]
Based on how the Scrapy project tests pipelines that return `Deferred`s, the callback for the `Deferred` should return the item. That suggests that the next pipeline is not invoked until the returned `Deferred` finishes.

**Comment by CarterPape** [comment]
Also: Duh, the [`process_item` method](https://docs.scrapy.org/en/latest/topics/item-pipeline.html#process_item) is expected to take an item as an argument — no mention in the documentation of accepting a `Deferred`.
